components/cards/ClaimAction.tsx

```diff
@@ -13,78 +21,172 @@ export type ClaimActionProps = SidebarProps & {
-  // }
-
-  // const { displayNewForSalePrice, displayNetworkFeePayment } = actionData;
+  const {
+    displayNewForSalePrice,
+    // displayNetworkFeePayment
+  } = actionData;
 
   // React.useEffect(() => {
-  //   let _transactionSubtotal = calculateWeiSubtotalField(
+  //   const _transactionSubtotal = calculateWeiSubtotalField(
   //     displayNetworkFeePayment
   //   );
 
   //   updateActionData({ transactionSubtotal: _transactionSubtotal });
   // }, [displayNetworkFeePayment]);
 
   async function _claim() {
-    // FIXME: Update for streaming payments
-    // updateActionData({ isActing: true });
-    // let baseCoord = GeoWebCoordinate.make_gw_coord(
-    //   claimBase1Coord.x,
-    //   claimBase1Coord.y
-    // );
-    // let destCoord = GeoWebCoordinate.make_gw_coord(
-    //   claimBase2Coord.x,
-    //   claimBase2Coord.y
-    // );
-    // let path = GeoWebCoordinate.make_rect_path(baseCoord, destCoord).map(
-    //   (v) => {
-    //     return BigNumber.from(v.toString(10));
-    //   }
-    // );
-    // if (path.length == 0) {
-    //   path = [BigNumber.from(0)];
-    // }
-    // const newValue = calculateWeiSubtotalField(displayNewForSalePrice);
-    // const newContributionRate = fromValueToRate(
-    //   newValue,
-    //   perSecondFeeNumerator,
-    //   perSecondFeeDenominator
-    // );
-    // const resp = await claimerContract.claim(
-    //   account,
-    //   baseCoord.toString(10),
-    //   path,
-    //   newContributionRate,
-    //   {
-    //     from: account,
-    //     value: calculateWeiSubtotalField(displayNetworkFeePayment),
-    //   }
-    // );
-    // const receipt = await resp.wait();
-    // const filter = claimerContract.filters.ParcelClaimed(null, null);
-    // const res = await claimerContract.queryFilter(
-    //   filter,
-    //   receipt.blockNumber,
-    //   receipt.blockNumber
-    // );
-    // let licenseId = res[0].args[0];
-    // return licenseId;
-    return "";
+    const baseCoord = GeoWebCoordinate.make_gw_coord(
+      claimBase1Coord.x,
+      claimBase1Coord.y
+    );
+    const destCoord = GeoWebCoordinate.make_gw_coord(
+      claimBase2Coord.x,
+      claimBase2Coord.y
+    );
+    let path = GeoWebCoordinate.make_rect_path(baseCoord, destCoord).map(
+      // FIXME: it should be 'BN' type, waiting for js-geo-web-coordinate upgrade.
+      (v: any) => {
+        return BigNumber.from(v.toString(10));
+      }
+    );
+    if (path.length == 0) {
+      path = [BigNumber.from(0)];
+    }
+
+    if (!displayNewForSalePrice) {
+      throw new Error(
+        `displayNewForSalePrice is invalid: ${displayNewForSalePrice}`
+      );
+    }
+
+    console.log(actionData);
+
+    let claimData;
+
+    let actionDataToPassInUserData;
+
+    let userData;
+
+    let txn;
+
+    const filter = claimerContract.filters.ParcelClaimed(null, null);
+    const parcels = await claimerContract.queryFilter(filter);
+    const existingParcels = parcels.length > 0;
```

Comment:
[2/2]

> If there are any existing parcels

Reply:
Does this check for any existing parcels from any user? You should be able to check if `existingFlow` below is > 0.